server/src/session.ts

```diff
@@ -873,42 +842,84 @@ export class Session {
   }
 
   /**
-   * Disable the language service if the specified `project` is not Angular or
-   * Ivy mode is enabled.
+   * Disable the language service of the specified `project` if its dependencies
+   * do not contain `@angular/core`.
+   *
+   * @returns main declaration file in `@angular/core`.
    */
-  private checkProject(project: ts.server.Project) {
+  private disableLanguageServiceIfNotAngular(project: ts.server.Project): string|undefined {
```

Comment:
How about `findAngularCoreOrDisableLanguageService`?

Reply:
sgtm, will rename and update docs.